Simplify expect_publish_diagnostics_error_count

samwgoldman opened the pull request:
Summary:
A few changes here:
1. Use the simpler expect_message util, since we always skip over unexpected messages
2. Match on the _parsed_ params type instead of directly on JSON
3. Skip apparently unnecessary path canonicalization (and remove redundant comments)
4. Only wait for the first publishDiagnostics request, then assert about it, instead of waiting for a publishDiagnostics request w/ the expected attributes

I validated (3) just by running tests locally. It's possible this will break when run on Mac/Win, but I'll check CI. The blame diff didn't explain this choice.

For (4), I think the proposed behavior is better, and tests continue to pass. In the scenario where we do expect multiple publishDiagnostics notifications, I think it's better to expect them instead of relying on skipping them. (I expected to need to fix up tests because of this change, but happily everything just worked out.)

Differential Revision: D87360101

kinto0 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

samwgoldman committed:
Simplify expect_publish_diagnostics_error_count (#1614)

Summary:

A few changes here:
1. Use the simpler expect_message util, since we always skip over unexpected messages
2. Match on the _parsed_ params type instead of directly on JSON
3. Skip apparently unnecessary path canonicalization (and remove redundant comments)
4. Only wait for the first publishDiagnostics request, then assert about it, instead of waiting for a publishDiagnostics request w/ the expected attributes

I validated (3) just by running tests locally. It's possible this will break when run on Mac/Win, but I'll check CI. The blame diff didn't explain this choice.

For (4), I think the proposed behavior is better, and tests continue to pass. In the scenario where we do expect multiple publishDiagnostics notifications, I think it's better to expect them instead of relying on skipping them. (I expected to need to fix up tests because of this change, but happily everything just worked out.)

Reviewed By: kinto0

Differential Revision: D87360101
